locator/token_metadata.hh

```diff
@@ -310,13 +316,32 @@ class shared_token_metadata {
     mutable_token_metadata_ptr _shared;
     token_metadata_lock_func _lock_func;
 
+    // We use this barrier during the transition to a new token_metadata version to ensure that the
+    // system stops using previous versions. Here are the key points:
+    //   * A new phase begins when a mutable_token_metadata_ptr passed to shared_token_metadata::set has
+    //   a higher version than the current one.
+    //   * Each shared_token_metadata::set call initiates an operation on the barrier. If multiple calls
+    //   have the same version, multiple operations may be initiated with the same phase.
```

Comment:
In what situation will there be multiple calls to `shared_token_metadata::set` with the same version?

Reply:
There is no requirement that every change of `token_metadata` should increment the version or be followed by `barrier_and_wait`, for some stages of topology change sm this can be skipped for efficiency.